### PR TITLE
fix(router): publish missing types

### LINKED ISSUE #10522: RouterState not documented as Tree<ActivatedRoute>
RouterState documentation doesn't reflect that it is a Tree<ActivatedRoute>, with all the methods that that includes - which I assume to be a very important part of the state.

Also, this is inconsistent with the `Usage` code segment, which includes usage of it as a Tree.

### LINKED ISSUE #17778: QueryParamsHandling documentation
## I'm submitting a
Documentation Issue or request.

## Current behavior
There is no useful documentation for queryParamsHandling. There is a single line use case under NavigationExtras but this gives no detail regarding the parameters available to use. As far as I can tell there is no other reference to this in the documentation. A query for "queryParamsHandling" at the top level of the angular documentation yields nil results.

## Expected behavior

1.  This option should be documented.
2. I should be able to find references to this option by searching in the documentation.

### PR DESCRIPTION
**Pull request by trotyl**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: closes #10522, closes #17778.

+ `Tree` type are used as the base of [`RouterState`](https://next.angular.io/api/router/RouterState);
+ ~~`InitialNavigation` is used in other public APIs like [`ExtraOptions`](https://next.angular.io/api/router/ExtraOptions#initialNavigation);~~ (See [comment](https://github.com/angular/angular/pull/19950#discussion_r150714651) below)
+ `QueryParamsHandling` is used in other public APIs like [`RouterLink`](https://next.angular.io/api/router/RouterLink#queryParamsHandling).

## What is the new behavior?

Add those missing types in entry typings file (as well as in docs).

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/router/src/router.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ import {Params, isNavigationCancelingError} from './shared';
 import {DefaultUrlHandlingStrategy, UrlHandlingStrategy} from './url_handling_strategy';
 import {UrlSerializer, UrlTree, containsTree, createEmptyUrlTree} from './url_tree';
 import {forEach} from './utils/collection';
-import {TreeNode, nodeChildrenAsMap} from './utils/tree';
+import {Tree, TreeNode, nodeChildrenAsMap} from './utils/tree';
```
Comment (jasonaden) [inline packages/router/src/router.ts]: Why this change?

File: packages/router/src/index.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export {CanActivate, CanActivateChild, CanDeactivate, CanLoad, Resolve} from './
 export {DetachedRouteHandle, RouteReuseStrategy} from './route_reuse_strategy';
 export {NavigationExtras, Router} from './router';
 export {ROUTES} from './router_config_loader';
-export {ExtraOptions, ROUTER_CONFIGURATION, ROUTER_INITIALIZER, RouterModule, provideRoutes} from './router_module';
+export {ExtraOptions, InitialNavigation, ROUTER_CONFIGURATION, ROUTER_INITIALIZER, RouterModule, provideRoutes} from './router_module';
```
Comment (jasonaden) [inline packages/router/src/index.ts]: `InitialNavigation` is deprecated. So we shouldn't be adding it to the public API.

**Review by jasonaden — CHANGES_REQUESTED** [review]

File: packages/router/src/index.ts (hunk 2 above)
Comment (trotyl) [reply packages/router/src/index.ts]: If so, we should remove it from [documentation](https://next.angular.io/api/router/ExtraOptions#initialNavigation), isn't it?

File: packages/router/src/router.ts (hunk 1 above)
Comment (trotyl) [reply packages/router/src/router.ts]: `Tree` is already provided in [RouterState doc](https://next.angular.io/api/router/RouterState#interface-overview) but the link is incorrect.

Comment (trotyl) [reply packages/router/src/router.ts]: Seems my mistake here, I guess I was trying to add it to export.

**Comment by trotyl** [comment]
@jasonaden Updated, would you mind adding an `aio: preview` for it?

**Comment by mgol** [comment]
@trotyl `InitialNavigation` is still not defined in the docs, see https://pr19950-95ba193.ngbuilds.io/api/router/ExtraOptions#initialNavigation in the preview.

File: packages/router/src/index.ts (hunk 2 above)
Comment (mgol) [reply packages/router/src/index.ts]: @jasonaden Is the whole type deprecated? I though just some of the options. `initialNavigation: 'disabled'` is required for Angular Upgrade so if that was deprecated we need a replacement.

**Comment by mgol** [comment]
@jasonaden ping? Router's `initialNavigation` is still not documented.

**Review by jasonaden — CHANGES_REQUESTED** [review]
Because these types don't inline well into AIO, we've recently been inlining the types for the public API. So we can still have the internal type (like `QueryParamsHandling`), but for the external type you would inline the `'merge' | 'preserve' | ''` wherever it appears in the public API.

This is something we're looking to address, but for now that's how they should be documented externally.

**Comment by mgol** [comment]
@jasonaden I see. Should those definitions be inlined manually or is there a way to generate it during the build?

I suppose it's the [`initialNavigation` param](https://angular.io/api/router/ExtraOptions#initialNavigation) that should have the type definition inlined then?

**Comment by jasonaden** [comment]
Yes, that's correct. Also I guess `QueryParamsHandling` though, right? Because you're adding it to the public API in this PR. Instead, we should not add it to the public API but instead inline the options where it currently appears in the public API. So in these spots (plus updating the comment docs):

https://github.com/angular/angular/blob/master/packages/router/src/router.ts#L119
https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link.ts#L102

**Comment by trotyl** [comment]
I guess it would better to block or close this PR for now rather than hacking everywhere, is it right?

**Comment by atscott** [comment]
Closing as `QueryParamsHandling` and `UrlTree` have since been added to the public API.

**Comment by atscott** [comment]
Actually, it looks like `Tree` should still be in the publicApi since it's a base class for `RouterState`. Would you mind opening a new PR for that rather than trying to rebase this one?
